impl/src/main/java/org/glassfish/soteria/identitystores/LdapIdentityStore.java

```diff
@@ -331,20 +354,30 @@ else if (searchScope == LdapSearchScope.SUBTREE) {
     }
 
     private LdapContext createDefaultLdapContext() {
-        return createLdapContext(
+        try {
+          return createLdapContext(
                 ldapIdentityStoreDefinition.url(),
                 ldapIdentityStoreDefinition.bindDn(),
                 ldapIdentityStoreDefinition.bindDnPassword());
+        } catch (Exception e) {
```

Comment:
This would be an opportunity to catch an exception, possibly figure out what the problem is (Bad URL? Bad DN or password?) based on the particular exception thrown, and then throw an `IdentityStoreConfigurationException` that is _not_ caught by the calling code, and propagates back up to the caller.

We have the most information about the problem right here. If we throw it, and then catch it in, e.g., `checkThroughSearch()`, it's much harder for `checkThroughSearch()` to handle it appropriately.